beetsplug/fetchart.py

```diff
@@ -317,13 +325,36 @@ def get(self, album, plugin, paths):
         """Return the Cover Art Archive and Cover Art Archive release group URLs
         using album MusicBrainz release ID and release group ID.
         """
+        candidate_url = ""
         if 'release' in self.match_by and album.mb_albumid:
-            yield self._candidate(url=self.URL.format(mbid=album.mb_albumid),
-                                  match=Candidate.MATCH_EXACT)
+            candidate_url = self.URL.format(mbid=album.mb_albumid)
+            match_type = Candidate.MATCH_EXACT
         if 'releasegroup' in self.match_by and album.mb_releasegroupid:
-            yield self._candidate(
-                url=self.GROUP_URL.format(mbid=album.mb_releasegroupid),
-                match=Candidate.MATCH_FALLBACK)
+            candidate_url = self.GROUP_URL.format(
+                            mbid=album.mb_releasegroupid)
+            match_type = Candidate.MATCH_FALLBACK
+
+        # Cover Art Archive API offers pre-resized thumbnails at several sizes.
+        # If the maxwidth config matches one of the already available sizes
+        # fetch it directly intead of fetching the full sized image and
+        # resizing it.
+        if plugin.maxwidth in self.VALID_THUMBNAIL_SIZES:
+            # Confirm that the thumbnail image exists, otherwise fall back to
+            # the default front image URL.
+            size_suffix = "-" + str(plugin.maxwidth)
+            candidate_thumbnail_url = candidate_url + size_suffix
+
+            with closing(self.request(candidate_thumbnail_url, method='HEAD',
+                         message=u'checking for thumbnail')) as resp:
+                try:
+                    resp.raise_for_status()
+                    self._log.debug(u'thumbnail available')
+                    candidate_url = candidate_thumbnail_url
+                except requests.exceptions.HTTPError:
+                    self._log.debug(u'thumbnail not available')
```

Comment:
There's a slightly nicer way in Python to run code only when an exception does not occur:
```suggestion
                try:
                    resp.raise_for_status()
                except requests.exceptions.HTTPError:
                    self._log.debug(u'thumbnail not available')
                else:
                    self._log.debug(u'thumbnail available')
                    candidate_url = candidate_thumbnail_url
```

At this point, this could also plausibly be `if requests.status != 200` or similar.